### PR TITLE
fixes & implement #refund

### LINKED ISSUE #3: Add refund to creditcard source
Implement refund (full/partial) method for creditcard source payment.

### PR DESCRIPTION
**Pull request by abarrak**
* This resolves #3. Not tested in live mode yet.

  
* About first commit: `fix nil reference error in InvalidRequestError`:
  
  **Context:** 
Some failed API's returned json data have the 'error' attribute empty (i.e. nil later on)
which causes a invocation of undefined method.
  ##### Steps to reproduce:
  - `refund api call` integrated within the wrapper revealed the problem when a payment has a status of `failed`.
  - Possibly other calls may produce the bug.

### CODE REVIEW

File: lib/moyasar/actions/refund.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+      module ClassMethods
+        def refund(id, attrs = {})
+          response = request(:post, refund_url(id), params: attrs)
+
+          new(response.body)
+        end
+      end
+
+      module InstanceMethods
```
Comment (ecleel) [inline lib/moyasar/actions/refund.rb]: We need to have refund instance method as well. So if I had payment object 

```ruby
payment = Payment.find('id')
payment.refund
```

Comment (abarrak) [inline lib/moyasar/actions/refund.rb]: Can be serialised too 😄 

```ruby
Payment.find('id').refund
```

File: test/moyasar_test.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@ def test_should_accept_api_key
 
   def test_request_should_throw_exception_if_key_is_nil
     stub_server_request(:payments, status: 401)
- 
+
     assert_raises Moyasar::AuthenticationError do
       Moyasar.request(:get, '/v1/payments')
```
Comment (abarrak) [inline test/moyasar_test.rb]: This is a spot for inspection when enhancing the test setup ..
Sometimes fails locally too.

File: lib/moyasar/actions/refund.rb (hunk 1 above)
Comment (ecleel) [inline lib/moyasar/actions/refund.rb]: Perfect! I liked how you done it. Do we need separate test for instance refund method or what we have was enough?

Comment (abarrak) [inline lib/moyasar/actions/refund.rb]: Tests are not aligned correctly with not refund, 
but the other resources like update and create.

The current for refund like them. Will work on tests next ..

File: moyasar.gemspec [hunk 3]
```diff
@@ -23,13 +23,13 @@ Gem::Specification.new do |spec|
...
+  spec.bindir        = 'bin'
+  spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
 
   spec.add_development_dependency 'bundler' # , "~> 1.10"
   spec.add_development_dependency 'rake' # , "~> 10.0"
   spec.add_development_dependency 'minitest'
-  
+
   spec.required_ruby_version = '>= 2.0.0'
```
Comment (ecleel) [inline moyasar.gemspec]: I think we need to change required_ruby_version to >= 2.1.10

**Comment by ecleel** [comment]
Great work @abarrak. Are we ready to merge?

**Comment by abarrak** [comment]
Yes, this is OK & concludes work on refund and other fixes.

**Pull request merged**